mirage/apt/read_apt_xml.py

```diff
@@ -987,16 +986,39 @@ def read_globalalignment_template(self, template, template_name, obs, prop_param
             else:
                 long_pupil = 'CLEAR'
 
-        # Repeat for the number of exposures + 1
-        for j in range(n_exp + 1):
+        for fgs_conf in ga_fgs_configs:
+            fgs_grps = fgs_conf.find(ns + 'Groups').text
+            fgs_ints = fgs_conf.find(ns + 'Integrations').text
+
+        # TODO read guider detector requirement from GuideStarID special requirement in this case
+        guider_det_num = 1  # PLACEHOLDER ONLY
+        fgs_subarr = "FGS{}_FULL".format(guider_det_num)
+
+        # Repeat for the number of exposures
+        for j in range(n_exp):
             # Add all parameters to dictionary
-            tup_to_add = (pi_name, prop_id, prop_title, prop_category,
-                          science_category, typeflag, mod, subarr, pdithtype,
-                          pdither, sdithtype, sdither, sfilt, lfilt,
-                          rpatt, grps, ints, short_pupil,
-                          long_pupil, grismval, coordparallel,
-                          i_obs, j + 1, template_name, 'NIRCAM', obs_label,
-                          target_name, tracking)
+
+            if j==2 or j==5:
+                # This is an FGS image as part of GA
+
+                # Add FGS exposure to the dictionary
+                tup_to_add = (pi_name, prop_id, prop_title, prop_category,
+                              science_category, typeflag, 'N/A', fgs_subarr, pdithtype,
+                              pdither, sdithtype, sdither, 'N/A', 'N/A',
+                              'FGSRAPID', fgs_grps, fgs_ints, 'N/A',
+                              'N/A', 'N/A', coordparallel,
+                              i_obs, j + 1, template_name, 'FGS', obs_label,
+                              target_name, tracking)
```

Comment:
Based on my sleuthing, I have come to the conclusion that previously, there were no FGS  observations in the GA template but now that there are, (due to this section?), yam.create_inputs in the test means that for the FGS observations the code enters the elif statement on line 551 of apt_inputs which then tries to find the guider information. The problem is that this APT file does not have any information about which guider to use. 

These are the options as I see them: A) we can add in a guide star special requirement for the GA observations with FGS as the instrument being used, or B) we can try to make it skip this or put in a dummy guider. 

To me, the option A is the best since it will actually be testing the code, specifically the changes that I made. 

The problem: I do not know how this file is generated and where to make the necessary changes. @mperrin or @bhilbert4, do either of you know how this file gets generated and if it's possible to edit it?

Reply:
Is it possible to open the xml file in APT? If so, I think that then re-exporting it (after adding a guide star special requirement if necessary) will probably put it in the current format. 

Will GA observations always have a guide star special requirement? If not then the new code should probably be modified to handle that.

Reply:
I was just able to open it so I can try to edit it. 

Yes, the GA observations must have a guide star special requirement.